archinstall/lib/mirrors.py

```diff
@@ -404,12 +405,18 @@ def get_mirror_regions(self) -> list[MirrorRegion]:
 		return available_mirrors
 
 	def load_mirrors(self) -> None:
+		# Skip if already loaded
+		if self._status_mappings is not None:
```

Comment:
I don't think this function should terminate, rather the calling side should conditionally call it so that any consumers can decide on their own. 
In this case only this call should be guarded https://github.com/archlinux/archinstall/blob/1faac77c0d476165a9dee1616274c1a2749c49db/archinstall/lib/mirrors.py?plain=1#L307